spec/ngsild_swagger.json

```diff
@@ -0,0 +1,1077 @@
+                        "schema": {
+                            "$ref": "ngsild_schema.json#/definitions/ApiErrorResponse"
+                        }
+                    },
+                    "404": {
+                        "description": "Not Found",
+                        "schema": {
+                            "$ref": "ngsild_schema.json#/definitions/ApiErrorResponse"
+                        }
+                    }
+                }
+            }
+        },
+        "/csources": {
+            "post": {
+                "description": "Registers a new context source within an NGSI-LD system",
+                "operationId": "registerCsource",
+                "parameters": [
+                    {
+                        "name": "CsourceRegistrationDescription",
+                        "description": "Payload body in the request contains a JSON-LD object which represents the context source registration that is to be created",
+                        "in":   "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "ngsild_csource_schema.json#/definitions/CsourceRegistrationDescription"
+                        }
+                    }
+                ],
+                "consumes": [
+                    "application/json"
+                ],
+                "responses": {
+                    "201": {
+                        "description": "Created",
+                        "headers": {
+                            "Location" : {
+                                "schema": {
+                                    "type": "string"
+                                },
+                                "description": "Contains the resource URI of the created context source registration resource"
+                            }
+                      }
+                    },
+                    "400": {
+                        "description": "Bad request",
+                        "schema": {
+                            "$ref": "ngsild_schema.json#/definitions/ApiErrorResponse"
+                        }
+                    },
+                    "409": {
+                        "description": "Already exists",
+                        "schema": {
+                            "$ref": "ngsild_schema.json#/definitions/ApiErrorResponse"
+                        }
+                    },
+                    "422": {
+                        "description": "Unprocessable Context Source Registration",
+                        "schema": {
+                            "$ref": "ngsild_schema.json#/definitions/ApiErrorResponse"
+                        }
+                    }
+                }
+            },
+            "get": {
+                "description": "Retrieve a set of context sources which matches a specific query from an NGSI-LD system",
+                "operationId": "queryCsources",
+                "parameters": [
```

Comment:
can we avoid duplication?

Reply:
they are the same parameters as those used to query entities so probably duplication can be avoided

Reply:
Pending